azurerm/internal/services/containers/kubernetes_cluster_data_source_test.go

```diff
@@ -450,6 +451,28 @@ func testAccDataSourceKubernetesCluster_addOnProfileRouting(t *testing.T) {
 	})
 }
 
+func TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGateway(t *testing.T) {
+	checkIfShouldRunTestsIndividually(t)
+	testAccDataSourceKubernetesCluster_addOnProfileRouting(t)
+}
+
+func testAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGateway(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_kubernetes_cluster", "test")
+	r := KubernetesClusterDataSource{}
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: r.addOnProfileIngressApplicationGatewayConfig(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("addon_profile.#").HasValue("1"),
+				check.That(data.ResourceName).Key("addon_profile.0.ingress_application_gateway.#").HasValue("1"),
+				check.That(data.ResourceName).Key("addon_profile.0.ingress_application_gateway.0.enabled").HasValue("true"),
+				check.That(data.ResourceName).Key("addon_profile.0.ingress_application_gateway.0.effective_application_gateway_id").Exists(),
+			),
```

Comment:
Can we check all the attributes exported here?

Reply:
Yes. Should be all set now.